userdocs/diagnostics/temporary-pointers.md

```diff
@@ -0,0 +1,64 @@
+# Temporary Pointers
+A temporary, or ephemeral, pointer in Swift is a pointer which is introduced by an implicit function argument conversion and is only valid for the lifetime of the function call it appears in. There are a few ways to create a temporary pointer:
+
+- Using an inout-to-pointer conversion by passing an argument with `&`:
+
+  ```swift
+  func foo(bar: UnsafePointer<Int>) { /*...*/ }
+  var x: Int = 42
+  foo(bar: &x)
+  ```
+
+  In the example above, the `bar` passed to `foo` is a temporary pointer to `x` which is only valid until `foo` returns.
+
+  Not all inout-to-pointer conversions result in a temporary pointer. Passing global variables and stored static properties with no observers inout can produce non-ephemeral pointers. If a global variable or stored static property is of a tuple or struct type, its stored members without observers may also be passed inout as non-epehemeral pointers.
```

Comment:
Nit: The global variables must be stored too. Perhaps it would be better to phrase it like "Passing global variables and static properties inout can produce non-ephemeral pointers, as long as they are stored and have no observers"?

Reply:
👍 , I like that phrasing